packages/flutter/lib/src/widgets/scrollbar.dart

```diff
@@ -386,3 +541,701 @@ class ScrollbarPainter extends ChangeNotifier implements CustomPainter {
+  ///
+  /// Cannot be null, defaults to [Duration.zero].
+  final Duration pressDuration;
+
+  @override
+  RawScrollbarState<RawScrollbar> createState() => RawScrollbarState<RawScrollbar>();
+}
+
+/// The state for a [RawScrollbar] widget, also shared by the [Scrollbar] and
+/// [CupertinoScrollbar] widgets.
+///
+/// Controls the animation that fades a scrollbar thumb in and out of view.
+///
+/// Provides defaults gestures for dragging the scrollbar thumb and tapping on the
+/// scrollbar track.
+class RawScrollbarState<T extends RawScrollbar> extends State<T> with TickerProviderStateMixin<T> {
+  double? _dragScrollbarAxisPosition;
+  ScrollController? _currentController;
+  Timer? _fadeoutTimer;
+  late AnimationController _fadeoutAnimationController;
+  final GlobalKey  _scrollbarPainterKey = GlobalKey();
+  bool _hoverIsActive = false;
+
+
+  /// Used to paint the scrollbar.
+  ///
+  /// Can be customized by subclasses to further customize scrollbar behavior by
+  /// overriding [updateScrollbar].
+  @protected
+  late final ScrollbarPainter scrollbarPainter;
+
+  /// The current value of the animation that fades the scrollbar in and out of
+  /// view.
+  ///
+  /// The animation is triggered by scroll events, fading the scrollbar into
+  /// view upon scrolling, and out of view upon stopping.
+  ///
+  /// If [isAlwaysShown] is true, the animation will not be triggered to fade out.
+  @protected
+  Animation<double> get fadeoutOpacityAnimation => _fadeoutOpacityAnimation;
+  late Animation<double> _fadeoutOpacityAnimation;
+
+  @override
+  void initState() {
+    super.initState();
+    _fadeoutAnimationController = AnimationController(
+      vsync: this,
+      duration: widget.fadeDuration,
+    );
+    _fadeoutOpacityAnimation = CurvedAnimation(
+      parent: _fadeoutAnimationController,
+      curve: Curves.fastOutSlowIn,
+    );
+    scrollbarPainter = ScrollbarPainter(
+      color: widget.thumbColor,
+      thickness: widget.thickness ?? _kScrollbarThickness,
+      fadeoutOpacityAnimation: fadeoutOpacityAnimation,
+    );
+  }
+
+  @override
+  void didChangeDependencies() {
+    super.didChangeDependencies();
+    updateScrollbarPainter();
+    _maybeTriggerScrollbar();
+  }
+
+  // Waits one frame and cause an empty scroll event (zero delta pixels).
+  //
+  // This allows the thumb to show immediately when isAlwaysShown is true.
+  // A scroll event is required in order to paint the thumb.
+  void _maybeTriggerScrollbar() {
+    WidgetsBinding.instance!.addPostFrameCallback((Duration duration) {
+      if (widget.isAlwaysShown) {
+        _fadeoutTimer?.cancel();
+        // Wait one frame and cause an empty scroll event.  This allows the
+        // thumb to show immediately when isAlwaysShown is true. A scroll
+        // event is required in order to paint the thumb.
+        widget.controller!.position.didUpdateScrollPositionBy(0);
+      }
+    });
+  }
+
+  /// Doc
+  @protected
+  void  updateScrollbarPainter() {
```

Comment:
I need to update docs still. @HansMuller is this more in line with the update feature we discussed? :)

Reply:
The changes look good. Would it be sufficient to call updateScrollbarPainter() from build and nowhere else?

Reply:
Updated! :)